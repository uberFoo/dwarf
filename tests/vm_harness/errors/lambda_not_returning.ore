fn main() -> int {
    let _0 = 0;
    let a = 42;
    let b = 96;
    let foo = |x: int, y: int| -> int {
        x + y + a;
    };
    let c = foo(1, 2);
    chacha::assert_eq(c, 45);
    c
}
