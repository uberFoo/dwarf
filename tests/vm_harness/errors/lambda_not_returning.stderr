[31mError:[0m Type mismatch: expected `[3;35mint[0m`, found `[3;35m()[0m`.
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m lambda_not_returning:6:15 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m5 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mf[0m[38;5;249mo[0m[38;5;249mo[0m[38;5;249m [0m[38;5;249m=[0m[38;5;249m [0m[38;5;249m|[0m[38;5;249mx[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mi[0m[38;5;249mn[0m[38;5;249mt[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249my[0m[38;5;249m:[0m[38;5;249m [0m[38;5;249mi[0m[38;5;249mn[0m[38;5;249mt[0m[38;5;249m|[0m[38;5;249m [0m[38;5;249m-[0m[38;5;249m>[0m[38;5;249m [0m[33mi[0m[33mn[0m[33mt[0m[33m [0m[33m{[0m
 [38;5;240m  â”‚[0m                                   [33mâ”€[0m[33mâ”€[0m[33mâ”¬[0m[33mâ”€[0m[33mâ”€[0m  
 [38;5;240m  â”‚[0m                                     [33mâ•°[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m[33mâ”€[0m expected [36m[3;35mint[0m[0m
 [38;5;246m6 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249mx[0m[38;5;249m [0m[38;5;249m+[0m[38;5;249m [0m[38;5;249my[0m[38;5;249m [0m[31m+[0m[31m [0m[38;5;249ma[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m               [31mâ”€[0m[31mâ”¬[0m  
 [38;5;240m  â”‚[0m                [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m found [36m[3;35m()[0m[0m
[38;5;246mâ”€â”€â”€â•¯[0m

[31mError:[0m missing function definition
   [38;5;246mâ•­[0m[38;5;246mâ”€[0m[38;5;246m[[0m lambda_not_returning:8:13 [38;5;246m][0m
   [38;5;246mâ”‚[0m
 [38;5;246m8 â”‚[0m [38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249m [0m[38;5;249ml[0m[38;5;249me[0m[38;5;249mt[0m[38;5;249m [0m[38;5;249mc[0m[38;5;249m [0m[38;5;249m=[0m[38;5;249m [0m[31mf[0m[31mo[0m[31mo[0m[38;5;249m([0m[38;5;249m1[0m[38;5;249m,[0m[38;5;249m [0m[38;5;249m2[0m[38;5;249m)[0m[38;5;249m;[0m
 [38;5;240m  â”‚[0m             [31mâ”€[0m[31mâ”¬[0m[31mâ”€[0m  
 [38;5;240m  â”‚[0m              [31mâ•°[0m[31mâ”€[0m[31mâ”€[0m[31mâ”€[0m function called here is not found
[38;5;246mâ”€â”€â”€â•¯[0m

